### PR TITLE
Chore/update sql side panel

### PR DESCRIPTION
**Pull request by MildTomato**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES/NO

## What kind of change does this PR introduce?

Bug fix, feature, docs update, ...

## What is the current behavior?

- Updated components to use `InnerSidePanelMenu` components
- Updated loading skeleton (see below)
  - stops layout shifting

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

- also updated skeleton loading state that looks more like SQL snippet items
Screenshot 2024-10-23 at 17 31 58

Deps on https://github.com/supabase/supabase/pull/30048

### CODE REVIEW

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx [hunk 1]
```diff
@@ -391,33 +397,36 @@ export const SQLEditorNav = ({ searchText: _searchText }: SQLEditorNavProps) =>
...
+            <InnerSideMenuCollapsibleContent className="group-data-[state=open]:pt-2">
               {numFavoriteSnippets === 0 ? (
-                <div className="mx-4">
-                  <Alert_Shadcn_ className="p-3">
-                    <AlertTitle_Shadcn_ className="text-xs">No favorite queries</AlertTitle_Shadcn_>
-                    <AlertDescription_Shadcn_ className="text-xs ">
+                <InnerSideBarEmptyPanel
+                  title="No favorite queries"
+                  className="mx-2"
+                  // @ts-expect-error - TODO: Fix this
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx:421]: just to check - what's the fix required here?

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx [hunk 2]
```diff
@@ -448,34 +457,61 @@ export const SQLEditorNav = ({ searchText: _searchText }: SQLEditorNavProps) =>
...
-                <AlertTitle_Shadcn_ className="text-xs">No queries created yet</AlertTitle_Shadcn_>
-                <AlertDescription_Shadcn_ className="text-xs">
-                  Queries will be automatically saved once you start writing in the editor.
-                </AlertDescription_Shadcn_>
-              </Alert_Shadcn_>
-            </div>
+            <InnerSideBarEmptyPanel
+              className="mx-3"
+              title="No queries created yet"
+              description="Something something something"
```
Comment (joshenlim) [inline apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx:503]: this looks wrong

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx [hunk 3]
```diff
@@ -448,34 +457,61 @@ export const SQLEditorNav = ({ searchText: _searchText }: SQLEditorNavProps) =>
...
+              actions={
+                <Button
+                  type="default"
+                  onClick={() => {
+                    // handle create query here
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx:508]: this is for you guys to fix. it wasn't clear how to do it

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx (hunk 1 above)
Comment (MildTomato) [reply apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx:421]: fixed

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx (hunk 2 above)
Comment (MildTomato) [reply apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx:503]: updated

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx]: realised this button isn't really relevant - the user's already on a page for a new snippet (given empty state 0 snippets), clicking on this button wouldn't do anything

**Comment by joshenlim** [comment]
@MildTomato one more style issue i noticed - something wonky with the collapsing animations here

https://github.com/user-attachments/assets/0ed63b76-2b57-42ba-944a-bc3bbdbc04cb

**Comment by MildTomato** [comment]
@joshenlim not sure what it is yet. it's affecting all Collapsibles when animated

File: apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/studio/components/layouts/SQLEditorLayout/SQLEditorNavV2/SQLEditorNav.tsx]: ok :D

**Comment by MildTomato** [comment]
We could remove the animation for now, until it's fixed. It's far more prominent here than other areas. it's annoying in this side bar as so much UI is flickering around

**Comment by MildTomato** [comment]
Removed the animation

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview

**Pull request merged**